test/autoupdate.test.js

```diff
@@ -0,0 +1,176 @@
+        if (err) return done(err);
+        // result should contain 'email', 'fullName_index', 'postcode'
+        // should not contain 'age', 'name_index'
+        Object.keys(result).length.should.equal(3);
+        EXPECTED_INDEXES = getExpectedIndexesForUpdatedModel();
+        async.eachOf(result, assertIndex, done);
+      });
+    });
+  });
+});
+
+function getTestModelDef() {
+  return {
+    properties: {
+      email: {type: 'string', index: true},
+      age: {type: 'number', index: true},
+      firstName: {type: 'string'},
+      lastName: {type: 'string'},
+    },
+    config: {
+      indexes: {
+        'name_index': {
+          keys: {
+            firstName: 1,
+            lastName: 1,
+          },
+        },
+      },
+    },
+  };
+}
+
+function getNewTestModelDef() {
+  return {
+    properties: {
+      email: {type: 'string', index: true},
+      age: {type: 'number'},
+      postcode: {type: 'string', index: true},
+      firstName: {type: 'string'},
+      middleName: {type: 'string'},
+      lastName: {type: 'string'},
+    },
+    config: {
+      indexes: {
+        'fullName_index': {
+          keys: {
+            firstName: 1,
+            middleName: 1,
+            lastName: 1,
+          },
+        },
+      },
+    },
+  };
+}
+
+function getExpectedIndexesForFirstCreatedModel() {
+  var result = {
+    age_index: {
+      ddoc: '_design/LBModel__AutoupdateTestFoo__LBIndex__age_index',
+      fields: [{age: 'asc'}],
+    },
+    email_index: {
+      ddoc: '_design/LBModel__AutoupdateTestFoo__LBIndex__email_index',
+      fields: [{email: 'asc'}],
+    },
+    name_index: {
+      ddoc: '_design/LBModel__AutoupdateTestFoo__LBIndex__name_index',
+      fields: [{firstName: 'asc'}, {lastName: 'asc'}],
+    },
+  };
+  return result;
+}
+
+function getExpectedIndexesForUpdatedModel() {
+  var result = {
+    postcode_index: {
+      ddoc: '_design/LBModel__AutoupdateTestFoo__LBIndex__postcode_index',
+      fields: [{postcode: 'asc'}],
+    },
+    email_index: {
+      ddoc: '_design/LBModel__AutoupdateTestFoo__LBIndex__email_index',
+      fields: [{email: 'asc'}],
+    },
+    fullName_index: {
+      ddoc: '_design/LBModel__AutoupdateTestFoo__LBIndex__fullName_index',
+      fields: [{firstName: 'asc'}, {lastName: 'asc'}, {middleName: 'asc'}],
+    },
+  };
+  return result;
+}
+
+function assertIndex(value, key, cb) {
+  EXPECTED_INDEXES[key].should.exist;
+  checkDdocname(key, value.ddoc);
+  checkFields(key, value.fields);
+  cb();
+}
+
+function checkDdocname(key, ddocName) {
+  EXPECTED_INDEXES[key].ddoc.should.equal(ddocName);
+}
+
+function checkFields(key, fields) {
+  arrayEqual(EXPECTED_INDEXES[key].fields, fields);
+};
+
+function arrayEqual(expect, actual) {
+  var notEqualMsg = util.inspect(expect, 4) + ' is not equal to ' +
+    util.inspect(actual, 4);
+  for (var item in expect) {
+    var key = Object.keys(expect[item])[0];
```

Comment:
I made a mistake here, the code should be 
```js
for (var item in expect) {
  var cond = expect[item];
  var i = _.findIndex(actual, cond);
  i.should.above(-1, notEqualMsg);
}
```
The same comparison logic is been fixed in `lib/migrate.js`, @ssh24 could you copy paste the code above and fix it in a new commit? Thanks!